File: src/main/java/com/range/rpms/users/api/AuthApi.java
package com.range.rpms.users.api;

import com.range.rpms.common.dto.GenericResponse;
import com.range.rpms.users.dto.UserLoginRequest;
import com.range.rpms.users.dto.UserRegisterRequest;
import io.swagger.v3.oas.annotations.Operation;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

@RequestMapping("/${app.base-path}/auth")
public interface AuthApi  {

    @Operation(
            summary = "resgister",
            description = "register to the app "
    )
    @PostMapping("/register")
    ResponseEntity<GenericResponse<String>> register(@RequestBody @Valid UserRegisterRequest userRegisterRequest);
    @Operation(
            summary = "login",
            description = "login to the app "
    )
    @PostMapping("/login")
    ResponseEntity<GenericResponse<String>> login(@RequestBody @Valid UserLoginRequest userLoginRequest);




}

File: src/main/kotlin/com/range/rpms/users/api/AuthApi.kt
package com.range.rpms.users.api

import com.range.rpms.users.dto.AuthResponse
import com.range.rpms.users.dto.ForgotPasswordRequest
import com.range.rpms.users.dto.LoginRequest
import com.range.rpms.users.dto.RegisterRequest
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam

@RequestMapping("/v1/api/auth")
interface AuthApi {
    @PostMapping("/login")
    fun login(@RequestBody loginRequest: LoginRequest): AuthResponse

    @PostMapping("/register")
    fun register(@RequestBody registerRequest: RegisterRequest): AuthResponse

    @PostMapping("/forgot-password")
    fun forgotPassword(@RequestBody forgotPasswordRequest: ForgotPasswordRequest)


}